store/src/java/com/zimbra/qa/unittest/TestCookieReuse.java

```diff
@@ -66,33 +68,43 @@
 import com.zimbra.soap.type.AccountSelector;
 import com.zimbra.soap.type.SearchHit;
 
-public class TestCookieReuse extends TestCase {
+public class TestCookieReuse {
+
+    @Rule
+    public TestName testInfo = new TestName();
+
     private static final String NAME_PREFIX = TestUserServlet.class.getSimpleName();
-    private static final String USER_NAME = "user1";
-    private static final String UNAUTHORIZED_USER = "unauthorized@example.com";
-    private int currentSupportedAuthVersion = 2;
-    @Override
+    private static String USER_NAME;
+    private static String UNAUTHORIZED_USER;
+    private int currentSupportedAuthVersion;
+
+    @Before
     public void setUp()
     throws Exception {
+        currentSupportedAuthVersion =
+            Provisioning.getInstance().getLocalServer().getLowestSupportedAuthVersion();
+
+        String prefix = NAME_PREFIX + "-" + testInfo.getMethodName().toLowerCase() + "-";
+        USER_NAME = prefix + "user1";
+        UNAUTHORIZED_USER = AccountTestUtil.getAddress(prefix + "unauthorized");
         cleanUp();
-        // Add a test message, in case the account is empty.
+        TestUtil.createAccount(USER_NAME);
         ZMailbox mbox = TestUtil.getZMailbox(USER_NAME);
+        // Add a test message, to make sure the account isn't empty
         TestUtil.addMessage(mbox, NAME_PREFIX);
-        if(TestUtil.accountExists(UNAUTHORIZED_USER)) {
-            TestUtil.deleteAccount(UNAUTHORIZED_USER);
-        }
     }
 
-    @Override
+    @After
     public void tearDown()
     throws Exception {
         cleanUp();
+        Provisioning.getInstance().getLocalServer().setLowestSupportedAuthVersion(currentSupportedAuthVersion);
```

Comment:
It sets the configuration back to how it was before the test, so other tests don't get nasty surprises.  The value is read during setup - if one of the tests that changes it were to assert before setting it back, that would be bad without this.